File: core/src/main/java/com/a65apps/core/interactors/contacts/PersonDetailsModel.java
package com.a65apps.core.interactors.contacts;

import com.a65apps.core.entities.Contact;
import com.a65apps.core.entities.Person;

import java.util.List;

import io.reactivex.rxjava3.annotations.NonNull;
import io.reactivex.rxjava3.core.Single;

public class PersonDetailsModel implements PersonDetailsInteractor {
    private final PersonDetailsRepository repository;

    public PersonDetailsModel(PersonDetailsRepository repository) {
        this.repository = repository;
    }


    @Override
    public Single<List<Contact>> loadContactsByPerson(@NonNull final String personId) {
        return this.repository.getContactsByPerson(personId);
    }

    @Override
    public Single<Person> loadPersonDetails(@NonNull String personId) {
        return this.repository.getPersonDetails(personId);
    }
}

File: core/src/main/java/com/a65apps/core/interactors/contacts/PersonDetailsModel.kt
package com.a65apps.core.interactors.contacts

import com.a65apps.core.entities.Contact
import com.a65apps.core.entities.Person
import kotlinx.coroutines.flow.Flow

class PersonDetailsModel(private val repository: PersonDetailsRepository) : PersonDetailsInteractor {
    override fun loadContactsByPerson(personId: String): Flow<List<Contact>> {
        return repository.getContactsByPerson(personId)
    }

    override fun loadPersonDetails(personId: String): Flow<Person> {
        return repository.getPersonDetails(personId)
    }
}
